### PR TITLE
manifest: openthread: regular openthread upmerge `37fb770`

### PR DESCRIPTION
**Pull request by canisLupus1313**
Implementation of new option:
`OPENTHREAD_OPERATIONAL_DATASET_AUTO_INIT`

### CODE REVIEW

File: modules/openthread/Kconfig.features [hunk 1]
```diff
@@ -212,6 +212,10 @@ config OPENTHREAD_NEIGHBOR_DISCOVERY_AGENT
 config OPENTHREAD_NETDATA_PUBLISHER
 	bool "Thread Network Data publisher"
 
+config OPENTHREAD_OPERATIONAL_DATASET_AUTO_INIT
+	bool "Operational dataset auto init"
+	default y
```
Comment (edmont) [inline modules/openthread/Kconfig.features:217]: OpenThread default is 0, should we default to `n` here?

Comment (canisLupus1313) [reply modules/openthread/Kconfig.features:217]: Doing so causes network forming fail. So Bsim tests won't pass.

**Review by pdgendt — DISMISSED** [review]

**Review by edmont — APPROVED** [review]

File: west.yml [hunk 2]
```diff
@@ -293,7 +293,7 @@ manifest:
       revision: c904a01d4a882bcbb39987e0e2ce5308f49ac7ad
       path: modules/lib/open-amp
     - name: openthread
-      revision: 6d557383e8bc8654f813674e8631c2e56b8e70d0
+      revision: pull/101/head
```
Comment (rlubos) [inline west.yml:296]: `37fb77098982d17555dd7a3f58832714bb9df56e`

Comment (canisLupus1313) [reply west.yml]: updated

**Review by rlubos — APPROVED** [review]

**Review by pdgendt — APPROVED** [review]

**Review by mariuszpos — APPROVED** [review]

**Pull request merged**